Minio server crashes during mc session resume.

I am trying to copy 10 GB from minio sever to local system. I had network interruption and on executing 
mc session resume i got this error.

`
minio@minio2:~/atul/datasets$ mc session resume YoPSmdce
mc: <ERROR> Failed to copy ‘http://127.0.0.1:9004/atul1/10gb’. Input reader closed pre-maturely. Expected ‘10737418240’ bytes, but only received ‘10298105856’ bytes.
mc: <ERROR> Session safely terminated. To resume session ‘mc session resume YoPSmdce’
`

```
To configure Minio Client:
    $ wget https://dl.minio.io/client/mc/release/linux-amd64/mc
    $ chmod 755 mc
    $ ./mc config host add myminio http://127.0.0.1:9004 3SNEB29QG2MJOGL8VY4F xz
b+SzZOfXUDTOFx5CwPZjzh2yy9Rdk2B8FYCLfG
panic: runtime error: slice bounds out of range

goroutine 1586489 [running]:
panic(0x9fae00, 0xc82000e0b0)
        /home/minio/.gimme/versions/go1.6.linux.amd64/src/runtime/panic.go:464 +
0x3e6
main.XL.ReadFile.func2(0xc8204d22d8, 0x1, 0x500000, 0xecec17d75, 0x17e3ec13, 0xf
b6920, 0x0, 0x5, 0x5, 0x400000, ...)
        /home/minio/gopath/src/github.com/minio/minio/xl-erasure-v1-readfile.go:
188 +0x1a39
created by main.XL.ReadFile
        /home/minio/gopath/src/github.com/minio/minio/xl-erasure-v1-readfile.go:
215 +0x9f1
```

--- a/xl-erasure-v1-readfile.go
+++ b/xl-erasure-v1-readfile.go
@@ -174,13 +174,14 @@ func (xl XL) ReadFile(volume, path string, startOffset int64) (io.ReadCloser, er
 
 			// Verify if the offset is right for the block, if not move to
 			// the next block.
+
 			if startOffset > 0 {
 				startOffset = startOffset - int64(len(dataBlocks))
-				if startOffset > int64(len(dataBlocks)) {
+				// Start offset is greater than or equal to zero, skip the dataBlocks.
+				if startOffset >= 0 {
 					continue
 				}
-				// Fetch back the overflow offset, to skip from the current data
-				// blocks.
+				// Now get back the remaining offset if startOffset is negative.
 				startOffset = startOffset + int64(len(dataBlocks))
 			}
 

